src/extrema.jl

```diff
@@ -0,0 +1,117 @@
+"""
+BlobLoG stores information about the location of peaks as discovered by `blob_LoG`.
+It has fields:
+
+- location: the location of a peak in the filtered image (a CartesianIndex)
+- σ: the value of σ which lead to the largest `-LoG`-filtered amplitude at this location
+- amplitude: the value of the `-LoG(σ)`-filtered image at the peak
+
+Note that the radius is equal to σ√2.
+
+See also: [`blob_LoG`](@ref).
+"""
+struct BlobLoG{T,S,N}
+    location::CartesianIndex{N}
+    σ::S
+    amplitude::T
+end
+
+"""
+    blob_LoG(img, σscales; edges=(true, false, ...), σshape=(1, ...)) -> Vector{BlobLoG}
+
+Find "blobs" in an N-D image using the negative Lapacian of Gaussians
+with the specifed vector or tuple of σ values. The algorithm searches for places
+where the filtered image (for a particular σ) is at a peak compared to all
+spatially- and σ-adjacent voxels, where σ is `σscales[i] * σshape` for some i.
+By default, `σshape` is an ntuple of 1s.
+
+The optional `edges` argument controls whether peaks on the edges are
+included. `edges` can be `true` or `false`, or a N+1-tuple in which
+the first entry controls whether edge-σ values are eligible to serve
+as peaks, and the remaining N entries control each of the N dimensions
+of `img`.
+
+# Citation:
+
+Lindeberg T (1998), "Feature Detection with Automatic Scale Selection",
+International Journal of Computer Vision, 30(2), 79–116.
+
+See also: [`BlobLoG`](@ref).
+"""
+function blob_LoG(img::AbstractArray{T,N}, σscales::Union{AbstractVector,Tuple};
+                  edges::Union{Bool,Tuple{Vararg{Bool}}}=(true, ntuple(d->false, Val(N))...), σshape=ntuple(d->1, Val(N))) where {T,N}
+    if edges isa Bool
+        edges = (edges, ntuple(d->edges,Val(N))...)
+    end
+    sigmas = sort(σscales)
+    img_LoG = Array{Float64}(undef, length(sigmas), size(img)...)
+    colons = ntuple(d->Colon(), Val(N))
+    @inbounds for isigma in eachindex(sigmas)
+        img_LoG[isigma,colons...] = (-sigmas[isigma]) * imfilter(img, Kernel.LoG(ntuple(i->sigmas[isigma]*σshape[i],Val(N))))
```

Comment:
It looks like we can use `imfilter!` here.

Reply:
That would destroy the original image, and we need to filter the original with a range of different `σ`. Or are you proposing that we first replicate and then filter in-place?

Reply:
I meant this:

```julia
    LoG_slice = @view img_LoG[isigma, colons...]
    imfilter!(LoG_slice, img, ...)
    LoG_slice .*= -sigmas[isigma]
```